src/logic/transaction-middleware.ts

```diff
@@ -0,0 +1,57 @@
+import { IHttpServerComponent } from '@well-known-components/interfaces'
+import { TransactionData } from '../ports/transaction/types'
+import { Schema } from '../ports/validation/types'
+import { AppComponents, Context } from '../types'
+
+export function createTransactionMiddleware(
+  components: Pick<AppComponents, 'logs' | 'validation' | 'transaction'>
+): IHttpServerComponent.IRequestHandler<Context<string>> {
+  const { logs, validation, transaction } = components
+  const logger = logs.getLogger('transaction-wrapper')
+
+  const transactionSchema: Schema<TransactionData> = {
+    type: 'object',
+    properties: {
+      from: { type: 'string' },
+      to: { type: 'string' },
+      params: { type: 'array', items: { type: 'string' } },
+    },
+    additionalProperties: false,
+    required: ['from', 'to', 'params'],
+  }
```

Comment:
This could be part of the `logic/transactions` namespace, a pure function that performs validations. To make smaller the handler testing surface

Reply:
will do 👍
 I didn't really know where to put it, the transaction component didn't feel like the right place